src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CaseRetriever.java

```diff
@@ -8,28 +15,101 @@
+
     private final CcdAuthenticatorFactory factory;
+
+    @LoadBalanced
     private final CoreCaseDataApi coreCaseDataApi;
 
     CaseRetriever(CcdAuthenticatorFactory factory, CoreCaseDataApi coreCaseDataApi) {
         this.factory = factory;
         this.coreCaseDataApi = coreCaseDataApi;
     }
 
-    public CaseDetails retrieve(String jurisdiction, String caseRef) {
-        CcdAuthenticator info = factory.createForJurisdiction(jurisdiction);
-        return retrieveCase(jurisdiction, caseRef, info);
+    public CcdAuthenticator authenticate(String jurisdiction) {
+        return factory.createForJurisdiction(jurisdiction);
     }
 
-    private CaseDetails retrieveCase(String jurisdiction, String caseRef, CcdAuthenticator authenticator) {
-        return coreCaseDataApi.readForCaseWorker(
+    @HystrixCommand(
+        commandKey = "case-retrieval",
+        commandProperties = {
+            @HystrixProperty(name = "execution.isolation.thread.timeoutInMilliseconds", value = "5000")
+        },
+        fallbackMethod = "retrieveFallback"
+    )
+    public CaseDetails retrieve(String jurisdiction, String caseRef, CcdAuthenticator authenticator) {
+        CaseDetails caseDetails = coreCaseDataApi.readForCaseWorker(
             authenticator.getUserToken(),
             authenticator.getServiceToken(),
             authenticator.userDetails.getId(),
             jurisdiction,
             CASE_TYPE_ID,
             caseRef
         );
+
+        if (caseDetails != null) {
+            log.info(
+                "Found worker case: {}:{}:{}",
+                caseDetails.getJurisdiction(),
+                caseDetails.getCaseTypeId(),
+                caseDetails.getId()
+            );
+        }
+
+        return caseDetails;
+    }
+
+    @SuppressWarnings("unused")
+    private CaseDetails retrieveFallback(
+        String jurisdiction,
+        String caseRef,
+        CcdAuthenticator authenticator,
+        Throwable exception
+    ) {
+        String message = exception.getMessage();
+
+        if (exception instanceof FeignException) {
+            HttpStatus status = HttpStatus.valueOf(((FeignException) exception).status());
+
+            logResponse(jurisdiction, caseRef, status, message);
+        } else {
+            log.error(
+                "Could not retrieve case with ref:{} for {} jurisdiction. Error: {}",
+                caseRef,
+                jurisdiction,
+                message,
+                exception
+            );
+        }
+
+        return null;
     }
 
+    private void logResponse(String jurisdiction, String caseRef, HttpStatus status, String originalMessage) {
+        if (HttpStatus.NOT_FOUND.equals(status)) {
+            log.info(
+                "Not found case ref:{} for {} jurisdiction {}. Error: {}",
+                caseRef,
+                jurisdiction,
+                originalMessage
+            );
+        } else if (status.is4xxClientError()) {
+            log.warn(
```

Comment:
I would think that any error back from CCD other than not found is an error, i.e. either we've done something wrong or a service team has.